### ISSUE TITLE
CMake buid scripts don't offer `--enable-keylog-export` like option

### ISSUE THREAD

**Issue by redbaron** [wolfSSL/wolfssl#8165]
### Version

5.7.4

### Description

configure script has `--enable-keylog-export` param which defines `WOLFSSL_SSLKEYLOGFILE`. This feature doesn't seem to be supported by CMake build scripts.

**Comment by dgarske** [comment]
@bigbrett can you review this bug report and evaluate effort to fix?

**Comment by bigbrett** [comment]
@redbaron Thanks for pointing that out, we can add it in.

In the meantime, if you ensure the C macros `-DSHOW_SECRETS -DHAVE_SECRET_CALLBACK -DWOLFSSL_SSLKEYLOGFILE -DWOLFSSL_KEYLOG_EXPORT_WARNED` are defined, it will have the equivalent behavior.

**Comment by redbaron** [comment]
@bigbrett , yeah passing `-DWOLFSSL_DEFINITIONS` with list of defines does the trick nicely in all cases so far, but first class CMake support will help discovering these features easier.

In general it seems that autotools has more love: 148 vs 34 commits this year so far.  

```
$ git log  CMakeLists.txt cmake |grep -F ' 2024 ' | wc -l
34

$ git log  configure.ac |grep -F ' 2024 ' | wc -l  
148
```

It is probably because autotools is more popular among embedded devs, but your build [instructions](https://www.wolfssl.com/documentation/manuals/wolfssl/chapter02.html) don't mention CMake either :(

**Comment by bigbrett** [comment]
@redbaron Yep I agree, our CMake coverage does often slip behind. wolfSSL has a long history with autotools, and for better or for worse, that is our default supported build system. So we do rely on the community for issues like this in keeping our CMake support up to date. Community PRs for CMake support are always appreciated :)

**Comment by redbaron** [comment]
@bigbrett ,

> Community PRs for CMake support are always appreciated :)

took opportunity to add more ergonomic helper to define build option: https://github.com/wolfSSL/wolfssl/pull/8174
